File: XPile 3-17/Programs/test_gc.js
// @dr memory = gc

class Node {
    constructor(value = 0, label = "") {
        this.value = value;
        this.label = label;
    }
}


function make_nodes(count) {
    let i = 0;
    let total = 0;
    while ((i < count)) {
        let n = new Node();
        n.value = i;
        n.label = "node";
        total = (total + n.value);
        i = (i + 1);
    }
    return total;
}

function main() {
    let result = make_nodes(100);
    console.log(result);
    return 0;
}

function __ul_sort(arr, n) { const sub = arr.slice(0, n).sort((a, b) => a < b ? -1 : a > b ? 1 : 0); for (let i = 0; i < n; i++) arr[i] = sub[i]; }

main();

File: XPile 3-17/Programs/test_gc.py
# @dr memory = gc

from dataclasses import dataclass, field

def __ul_sort(arr, n): arr[:n] = sorted(arr[:n])

class __ul_Exception(Exception):
    def __init__(self, type_name, message):
        self.type_name = type_name
        self.message = message
        super().__init__(message)

class Node:
    def __init__(self, value=0, label=""):
        self.value = value
        self.label = label


def make_nodes(count):
    i = 0
    total = 0
    while (i < count):
        n = Node()
        n.value = i
        n.label = 'node'
        total = (total + n.value)
        i = (i + 1)
    return total

def main():
    result = make_nodes(100)
    print(result)
    return 0


if __name__ == "__main__":
    main()
